system/Commands/Utilities/Optimize.php

```diff
@@ -99,32 +123,100 @@ private function removeFile(string $cache): void
         }
     }
 
-    private function enableCaching(): void
+    private function runCaching(?bool $enableConfigCache, ?bool $enableLocatorCache, ?bool $disable): void
     {
-        $publisher = new Publisher(APPPATH, APPPATH);
+        // Prepare search and replace mappings
+        $searchReplace = [];
 
-        $config = APPPATH . 'Config/Optimize.php';
+        if ($disable === true) {
+            $searchReplace = $this->disableCaching();
+        } else {
+            $searchReplace = $this->enableCaching(['config' => $enableConfigCache, 'locator' => $enableLocatorCache]);
+        }
 
-        $result = $publisher->replace(
-            $config,
-            [
-                'public bool $configCacheEnabled = false;'  => 'public bool $configCacheEnabled = true;',
-                'public bool $locatorCacheEnabled = false;' => 'public bool $locatorCacheEnabled = true;',
-            ]
-        );
+        // Apply replacements if necessary
+        if ($searchReplace !== []) {
+            $publisher = new Publisher(APPPATH, APPPATH);
 
-        if ($result) {
-            CLI::write(
-                'Config Caching and FileLocator Caching are enabled in "app/Config/Optimize.php".',
-                'green'
-            );
+            $result = $publisher->replace(self::CONFIG_PATH, $searchReplace);
 
-            return;
+            if ($result === true) {
+                $messages = [];
+
+                if (in_array('public bool ' . self::CONFIG_CACHE . ' = true;', $searchReplace, true)) {
+                    $messages[] = 'Config Caching is enabled in "app/Config/Optimize.php".';
+                }
+
+                if (in_array('public bool ' . self::LOCATOR_CACHE . ' = true;', $searchReplace, true)) {
+                    $messages[] = 'FileLocator Caching is enabled in "app/Config/Optimize.php".';
+                }
+
+                if (in_array('public bool ' . self::CONFIG_CACHE . ' = false;', $searchReplace, true)) {
+                    $messages[] = 'Config Caching is disabled in "app/Config/Optimize.php".';
+                }
+
+                if (in_array('public bool ' . self::LOCATOR_CACHE . ' = false;', $searchReplace, true)) {
+                    $messages[] = 'FileLocator Caching is disabled in "app/Config/Optimize.php".';
+                }
+
+                foreach ($messages as $message) {
+                    CLI::write($message, 'green');
+                    CLI::newLine();
+                }
+
+                CLI::newLine();
+
+                return;
+            }
+
+            CLI::error('Error in updating file: ' . clean_path(self::CONFIG_PATH));
+
+            throw new RuntimeException(__METHOD__);
         }
+        CLI::write('No changes to caching settings.', 'yellow');
+    }
 
-        CLI::error('Error in updating file: ' . clean_path($config));
+    /**
+     * Disable Caching
+     *
+     * @return array<string, string>
+     */
+    private function disableCaching(): array
```

Comment:
When I see the method name `disableCaching()`, I imagine the method disables caching.
But  this method just returns an array.
Therefore the  method name or the implementation is not good.

Reply:
i will send an update, this makes it more flexible.